src/Uno.UI/UI/Xaml/Controls/Popup/PlacementPopupPanel.cs

```diff
@@ -109,13 +109,60 @@ protected virtual Rect CalculateFlyoutPlacement(Size desiredSize, Size maxSize)
 			desiredSize.Width = Math.Min(desiredSize.Width, visibleBounds.Width);
 			desiredSize.Height = Math.Min(desiredSize.Height, visibleBounds.Height);
 
-			if (PositionInAnchorControl is Point point)
+			if (FlyoutShowOptions?.Position is { } position)
```

Comment:
Try to refactor this code into the FlyoutPopupPanel class, since it's not used for Tooltip at all.